src/cargo/util/toml/mod.rs

```diff
@@ -655,6 +655,20 @@ impl TomlProfile {
             }
         }
 
+        if let Some(lto) = &self.lto {
+            if let StringOrBool::String(arg) = lto {
+                // try as I like I could not get `Deref` pattern matching to work here
+                if arg == "true" || arg == "false" {
+                    bail!(
+                        "`lto` setting of string `\"{}\"` is not a valid setting, \
+                        must be a boolean (`true`/`false`) or a string (`\"yes\"`/ \
+                        `\"no\"`/`\"on\"`/`\"off\"`/`\"thin\"`/`\"fat\"`, etc) or omitted.",
```

Comment:
While this does match the output of rustc if you put `lto = "true"` it doesn't match the current [cargo lto docs](https://doc.rust-lang.org/cargo/reference/profiles.html?highlight=lto#lto). I think it might be a good idea to either update the current cargo docs to match [rustc](https://doc.rust-lang.org/rustc/codegen-options/index.html#lto) or to only include the things that the current cargo lto docs do. I would lean towards only keeping what cargo currently does as it differs greatly from the rustc docs. I would hold off on implementing this until a maintainer chimes in but I think it would be a mistake to not have the error message here match the documentation about lto.

Reply:
Thanks for the quick review, will do